### PR TITLE
[5.2] Added full URL is check

### PR DESCRIPTION
**Pull request by edcs**
Currently if you run a test on whether or not a URL matches a pattern using `Illuminate\Http\Request::is()` the query string is ignored. It would be useful to be able to include the query string in this test since `/foo` != `/foo?bar=123`.

This PR proposes a new method called `fullIs()` method in `Illuminate\Http\Request` which lets you compare the full URL with query string to a pattern or patterns, similar to the way the existing `is()` method works.

### CODE REVIEW

File: src/Illuminate/Http/Request.php [hunk 1]
```diff
@@ -165,6 +165,25 @@ public function is()
     }
 
     /**
+     * Determine if the current request URL and query string matches a pattern.
+     *
+     * @param  mixed  string
+     * @return bool
+     */
+    public function fullUrlIs()
```
Comment (GrahamCampbell) [inline src/Illuminate/Http/Request.php]: `fullIs` would be a better name imo

**Comment by taylorotwell** [comment]
It should be `fullUrlIs`.

**Comment by GrahamCampbell** [comment]
@taylorotwell We don't have `urlIs` though? We just have `is`.

**Comment by edcs** [comment]
Think I agree with @taylorotwell actually - it's comparing the result from the `fullUrl()` method. There's a `url()` method which returns a different value, so you'd expect `urlIs()` to compare the value from `url()` and `fullUrlIs()` to compare the value from `fullUrl()`.

So should I add a `urlIs()` and a `fullUrlIs()` method?

**Comment by GrahamCampbell** [comment]
Just make the change @taylorotwell suggested. If you want to change anything else, that falls outside the scope of this PR. ;)

**Pull request merged**